### PR TITLE
crypto/tls: allow the usage of other keyUsage when checking for the d…

### LINKED ISSUE #72: crypto/tls: keyUsage tests exclusively for DigitalSignature flag
The `tls.Config.KeyUsage` field can contain several flags. 
Currently, for generating a delegated credential extension, it tests that the `KeyUsageDigitalSignature` flags is enabled exclusively. 

https://github.com/cloudflare/go/blob/5ef1b90573f8742b763b7e65a43ce7fa20e37bb4/src/crypto/x509/x509.go#L1789-L1795

Shouldn't this 
`(template.KeyUsage == KeyUsageDigitalSignature)` 
be replaced by 
`(template.KeyUsage &  KeyUsageDigitalSignature != 0)`

i.e., only testing the flag is present, without baning from setting other flags in the same field.

### PR DESCRIPTION
**Pull request by claucece**
Should fix #72

### CODE REVIEW

File: src/crypto/x509/x509.go [hunk 1]
```diff
@@ -1787,13 +1787,15 @@ func buildCertExtensions(template *Certificate, subjectIsEmpty bool, authorityKe
 	}
 
 	// This extension is not critical
-	if template.KeyUsage == KeyUsageDigitalSignature && template.AllowDC && !template.IsCA && !oidInExtensions(oidExtensionDelegatedCredential, template.ExtraExtensions) {
-		ret[n].Id = oidExtensionDelegatedCredential
-		ret[n].Value, err = asn1.Marshal(asn1.NullRawValue)
-		if err != nil {
-			return
+	if template.AllowDC && !template.IsCA && !oidInExtensions(oidExtensionDelegatedCredential, template.ExtraExtensions) {
+		if template.KeyUsage&KeyUsageDigitalSignature != 0 {
```
Comment (cjpatton) [inline src/crypto/x509/x509.go]: Why next this if statement? I think it's clearer if you move the condition into the if statement above.

**Review by cjpatton — CHANGES_REQUESTED** [review]

File: src/crypto/x509/x509.go (hunk 1 above)
Comment (claucece) [reply src/crypto/x509/x509.go]: Solved.

File: src/crypto/x509/x509.go [hunk 2]
```diff
@@ -1787,7 +1787,7 @@ func buildCertExtensions(template *Certificate, subjectIsEmpty bool, authorityKe
 	}
 
 	// This extension is not critical
-	if template.KeyUsage == KeyUsageDigitalSignature && template.AllowDC && !template.IsCA && !oidInExtensions(oidExtensionDelegatedCredential, template.ExtraExtensions) {
+	if template.AllowDC && !template.IsCA && !oidInExtensions(oidExtensionDelegatedCredential, template.ExtraExtensions) && template.KeyUsage&KeyUsageDigitalSignature != 0 {
```
Comment (lukevalenta) [inline src/crypto/x509/x509.go]: Perhaps add some parentheses here to improve readability for the bit operation? `...&& (template.KeyUsage & KeyUsageDigitalSignature != 0) {`

**Review by lukevalenta — APPROVED** [review]
LGTM other than a comment to improve readability.

**Review by cjpatton — APPROVED** [review]

File: src/crypto/x509/x509.go (hunk 2 above)
Comment (claucece) [reply src/crypto/x509/x509.go]: Added now.

**Pull request merged**